internal/pkg/manifest/validate.go

```diff
@@ -1937,36 +1948,54 @@ func populateNLBPortsAndValidate(containerNameFor map[uint16]string, opts valida
 		return nil
 	}
 	nlb := opts.nlb
-	if nlb.Port == nil {
+	if nlb.MainListener.Port == nil {
 		return nil
 	}
-	nlbPort, _, err := ParsePortMapping(nlb.Port)
+
+	for _, listener := range nlb.NLBListeners() {
+		if err := populateAndValidateNLBPorts(listener, containerNameFor, opts); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func populateAndValidateNLBPorts(listener NetworkLoadBalancerListener, containerNameFor map[uint16]string, opts validateExposedPortsOpts) error {
+	nlbPort, _, err := ParsePortMapping(listener.Port)
```

Comment:
qq: is it possible for `listener.Port` to be `nil` here? I'm assuming we would have errored out if a listener has a `nil` port, is this correct?

Reply:
yes we error out if port is empty even before coming here. So port being nil here is not possible.